modules/tokenPage/component/MyAssetCard.tsx

```diff
@@ -0,0 +1,102 @@
+import { FunctionComponent, useEffect } from "react";
+import { useLeveragedTokenNAV } from "../../../components/v1/swr/useLeveragedTokenNAV";
+import { useTokenBalance } from "../../../components/v1/swr/useTokenBalance";
+import { useVaultExchangeRate } from "../../../components/v1/swr/useVaultExchangeRate";
+import { useWalletContext } from "../../../components/v1/Wallet";
+import ButtonTertiary from "../../../uikit/button/ButtonTertiary";
+import Button from "../../../uikit/buttonV2/Button";
+import InformationCard from "../../../uikit/card/InformationCard";
+import { addTokenToMetamask, tokenType } from "../../../utils/addTokenToMetamask";
+import { useAssetStore } from "../store/assetStore";
+import { LoadedData, useTokenStore } from "../store/tokenStore";
+import { AssetsItem } from "./AssetItem";
+
+type MyAssetsCardProps = {
+    isVault?: boolean;
+};
+
+const MyAssetsCard = ({ state, isVault }: { state: LoadedData; isVault: boolean }) => {
+    const { chainId, token: metadata, tokenId } = state;
+    const { setValue, setError, state: assetState, setLoading } = useAssetStore();
+
+    const { account, provider } = useWalletContext();
+    const navResponse = useLeveragedTokenNAV({ token: tokenId, vault: metadata.vaultAddress, provider: provider }, { isPaused: () => !tokenId || !metadata || !provider });
+    const latestVaultExchangeRateResponse = useVaultExchangeRate({ vault: metadata.vaultAddress, provider: provider }, { isPaused: () => !metadata || !provider });
+    const balanceResponse = useTokenBalance({ account: account, token: isVault ? metadata.vaultAddress : tokenId, provider: provider }, { isPaused: () => !tokenId || !metadata || !account });
+
+    const refetchData = () => {
+        setLoading();
+        navResponse.mutate();
+        latestVaultExchangeRateResponse.mutate();
+        balanceResponse.mutate();
+    };
+
+    useEffect(() => {
```

Comment:
Why we have multiples `useEffect()` ?

Reply:
because i think if you combine them , it may create unexpected re-render and side effect